### DISCUSSION TITLE
Custom https gitlab server. token not found

### DISCUSSION THREAD

**Discussion by Artawower** [magit/forge#386] [Support] [answered]
Hello! I used forge for gitlab.com, and its worked perfect. But now i'm trying to connect private gitlab server, but i faced error 
```
ghub--token: Required Gitlab token ("user^forge" for "git.domain-soft.com/api/v4") does not exist.
See https://magit.vc/manual/ghub/Getting-Started.html
or (info "(ghub)Getting Started") for instructions.
```

I made ~/.authinfo file with such content:
```
machine git.domain-soft.com/api/v4 login My_Nickname^forge password some-token
```

My config
```
(use-package forge
  :after magit
  :config
  ;; (setq auth-sources '((:source "~/.authinfo")))
  (setq auth-sources "~/.authinfo")
  ;; (setq auth-source-debug trivia)
  (push '("git.domain-soft.com" "git.domain-soft.com/api/v4"
          "git.domain-soft.com" forge-gitlab-repository)
        forge-alist))

```
What could i have messed? I use osx

**Comment by tarsius** [comment, marked as answer]
`(setq auth-sources '("~/.authinfo"))` and then restart to be sure all caches are gone.

**Reply by Artawower** [reply]
Same error

**Reply by tarsius** [reply]
Either this is the problem or you only broke this during obfuscation but the strings `user^forge` and `My_Nickname^forge` should obviously not differ from each other.

**Reply by Artawower** [reply]
Thanks a lot! I don't now what was the reason of this problem, but i didn't obfuscate error message. I only changed "doman" in first code snippet. But today this started work without any new changes (i restarted emacs yesterday after your first advice).  It's realy strange, cause i spent a lot of time yesterday to try fix this error. Should i mark your advice as decision?

**Reply by tarsius** [reply]
Glad it works now. Yeah, "accept" this "answer" to mark the whole thing as resolved.

**Comment by dcunited001** [comment]
This helped. I didn't realize the machine name should contain the API endpoint.
